docs/architecture/glossary.rst

```diff
@@ -7,66 +7,129 @@ Glossary
   BEL
   basic logic element
   BLE
-    For example a LUT5, LUT6, CARRY4, or MUX, but not PIPs.
+    Basic elements (BELs) or basic logic element (BLEs)
+    are the basic logic units in an FPGA, including
+    carry or fast adders (CFAs), flip flops (FFs), lookup tables (LUTs),
+    multiplexers (MUXes), and other element types.
+    Note: Programmable interconnects (PIPs) are not counted as BELs.
 
-    BELs come in two types:
+    BELs come in two forms:
 
     * Basic BEL - A logic unit which does things.
-    * Routing BEL - A unit which is statically configured at the routing time.
+    * Routing BEL - A unit which is statically configured at routing time.
      
 
-  bitstream
+  Bitstream
     Binary data that is directly loaded into an FPGA to perform configuration.
     Contains configuration :term:`frames <frame>` as well as programming
     sequences and other commands required to load and activate same.
 
-  clock domain
+  Clock domain
     Portion of a :term:`horizontal clock row` to one side of the global clock
     spine. Often refers to :term:`tiles <tile>` that are associated with these
     clocks.
 
-  column
-    Collection of :term:`tiles <tile>` physically organized as a vertical line.
+  Column
+    A term used in :term:`bitstream` configuration to denote
+    a collection of :term:`tiles <tile>`, physically organized as
+    a vertical line, and configured by the same set of configuration frames.
+    Logic columns span 50 tiles vertically and 2 tiles horizontally
+    (pairs of logic tiles and interconnect tiles).
 
-  configurable logic block
   CLB
-    Basic building block of logic. 
-
-  frame
-    Fundamental unit of configuration data consisting of 101 :term:`words <word>`.
-
-  half
+  Configurable logic block (CLB)
```

Comment:
This form makes the parenthesized initialism part of the glossary term used by other pages to link to this entry.  I'm unclear on what a good pattern is for this.  Do we restate the term and initialism in the definition?  Do we expect the reader to understand that the two term entries are equivalent?

Reply:
Good point. I've moved the abbreviation out of the term, so other pages don't need to include the abbreviation if they want to link to the full term. I ran the html and link checkers after making the change - all good.

On the pattern: I do think we should restate the term and the abbreviation in the definition when they're not obvious. We want to be as clear as possible, and also to tax the reader's internal parser as little as possible. It's not necessarily clear to the reader that a sequence of bold terms, each on a different row (BLE, Basic Logic Element, BEL, etc) are all the same thing, or that a particular phrase is the expansion of a given abbreviation, or that the next available definition applies to them all. Case in point: When I first came across this glossary, I thought we had a bunch of terms without definitions.

Another interesting point this thread raises, is the syntactical pattern of the definitions. If we repeat the term, we're forced to use a full sentence in the first part of the definition. Like: "A configurable logic block (CLB) is the configurable logic unit of an FPGA," rather than just a phrase: "The configurable logic unit of an FPGA."

I had a look at a few glossaries, especially those on Read the Docs sites, and they merrily mix and match full sentences versus phrases. I think it's OK for us to do that too, provided our goal is clarity and ease of comprehension.